demos/dotnet/dotnet/CscAction.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class CscAction
+    {
+        public static bool Execute(ProjectProperties properties, Log log)
+        {
+            Console.WriteLine("compiling");
+            var processSettings = new ProcessStartInfo
+            {
+                FileName = properties.CscPath,
+                Arguments = properties.FormatCscArguments()
+            };
+
+            log.WriteLine("Executing {0}", processSettings.FileName);
+            log.WriteLine("Csc Arguments: {0}", processSettings.Arguments);
+
+            processSettings.CreateNoWindow = true;
```

Comment:
Why are these properties not set above when you create processSettings? Seems like the initializer can own this concern all-up...